### ISSUE TITLE
xSQLServerSetup Missing system administrator account error

### ISSUE THREAD

**Issue by AlexandrosIV** [PowerShell/xSQLServer#116]
Hi, I keep getting  an error trying to install SQL using 

 xSQLServerSetup StandAloneInstall {
            InstanceName = 'SPFarm'
            SetupCredential = (Get-Credential sparta\sp_admin)
            Features =   'SQLEngine,FullText,SSMS'
            SourcePath  = '\\MYWINDOWSVM01\Binaries\SQL2014SP2\'
}

I get Exit message:   Missing system administrator account. To continue, provide at least one Windows account to provision as a SQL Server system administrator.

from the xSQLServerSetup resource code, I can see that by default it should use the the SetupCredential param but not sure why is that not happening
if($Features.Contains("SQLENGINE"))
    {
        $Arguments += " /SQLSysAdminAccounts=`"" + $SetupCredential.UserName + "`""
        if($PSBoundParameters.ContainsKey("SQLSysAdminAccounts"))
        {
            foreach($AdminAccount in $SQLSysAdminAccounts)
            {
                $Arguments += " `"$AdminAccount`""
            }
        }
        if($SecurityMode -eq "SQL")
        {
            $Arguments += " /SAPwd=" + $SAPwd.GetNetworkCredential().Password
        }
    }

in the Set block I can see the call to the installer with the arguments, but not the argument for SQLSysAdminAccounts
VERBOSE: [MYWINDOWSVM02]:                            [[xSQLServerSetup]StandAloneInstall] Path: \\MYWINDOWSVM01\Binaries\SQL2014SP2\Source\setup.exe
VERBOSE: [MYWINDOWSVM02]:                            [[xSQLServerSetup]StandAloneInstall] Arguments: /Quiet="True" /IAcceptSQLServerLicenseTerms="True" /Actio
n="Install" /InstanceName="SPFARM" /Features="SQLEngine,FullText,SSMS"

any idea what could I been doing wrong? the MOF is set to allow plain credentials and are stored and are valid  #credentials

**Comment by aultt** [comment]
You need to provide the SQL Server System Administraters as well as a few other parameters. Take a look at the examples folder dscsqlbuild.ps1 specifically for a good stand alone example

**Comment by AlexandrosIV** [comment]
since it not marked as required I thought it was not necessary.
nevertheless, I tried with the param and failing with same error.
The first time I tried with the resource worked fine, so probably at this point  will do a clean server install and write everything from scratch, but wanted to know what could be wrong with this

**Comment by johlju** [comment]
As you said, It is not required and does use the parameter `SetupCredential` as the first SQLSysAdminAccounts, but since you get an error apparently it does not set, so I guess there  is a bug here. Might be that `$SetupCredential.UserName` should be `$($SetupCredential.UserName)` (I have not tested). Otherwise the workaround is to set SQLSysAdminAccounts in addition to `SetupCredential`. 

```
...
    if($Features.Contains("SQLENGINE"))
    {
        $Arguments += " /SQLSysAdminAccounts=`"" + $($SetupCredential.UserName) + "`""
        if($PSBoundParameters.ContainsKey("SQLSysAdminAccounts"))
        {
            foreach($AdminAccount in $SQLSysAdminAccounts)
            {
                $Arguments += " `"$AdminAccount`""
            }
        }
...
``` 

If this is a problem with other fields not being required but actually are required then I fell that is a bug too and should be fixed. Let us know what fields should be made required so that can be fixed.

**Comment by AlexandrosIV** [comment]
I tried with different versions of SQL (2012, 2014 and 2016) and I am seeing that the parameters in the resource are not being passed down in the call to the installer
 
 ```
$cred = Get-Credential sparta\sp_admin
      
xSQLServerSetup StandAloneInstall {

            InstanceName = 'SPFarm'
            SetupCredential = $cred
            Features =   'SQLEngine,FullText'
            SourcePath  = '\\MYWINDOWSVM01\Binaries\SQL2012SP2\'
            SQLSvcAccount = $cred
            SQLSysAdminAccounts = 'sparta\sp_admin'
            AgtSvcAccount = $cred
            **ASSvcAccount = $cred**  
        }

```
but in the summary.txt with the errors, I can see some parameters set as in the resource, but all accounts parameters are empty, even though I set several of them

User Input Settings:
  ACTION:                        Install
  ADDCURRENTUSERASSQLADMIN:      false
  AGTSVCACCOUNT:                 NT Service\SQLAgent$SPFARM
  AGTSVCPASSWORD:                *****
  AGTSVCSTARTUPTYPE:             Manual
  ASBACKUPDIR:                   Backup
  ASCOLLATION:                   Latin1_General_CI_AS
  ASCONFIGDIR:                   Config
  ASDATADIR:                     Data
  ASLOGDIR:                      Log
  ASPROVIDERMSOLAP:              1
  ASSERVERMODE:                  MULTIDIMENSIONAL
  **ASSVCACCOUNT:                  <empty>**
  ASSVCPASSWORD:                 <empty>
  ASSVCSTARTUPTYPE:              Automatic
  ASSYSADMINACCOUNTS:            <empty>
  ASTEMPDIR:                     Temp
  BROWSERSVCSTARTUPTYPE:         Automatic
  CLTCTLRNAME:                   <empty>
  CLTRESULTDIR:                  <empty>
  CLTSTARTUPTYPE:                0
  CLTSVCACCOUNT:                 <empty>
  CLTSVCPASSWORD:                <empty>
  CLTWORKINGDIR:                 <empty>
  COMMFABRICENCRYPTION:          0
  COMMFABRICNETWORKLEVEL:        0
  COMMFABRICPORT:                0
  CONFIGURATIONFILE:             
  CTLRSTARTUPTYPE:               0
  CTLRSVCACCOUNT:                <empty>
  CTLRSVCPASSWORD:               <empty>
  CTLRUSERS:                     <empty>
  ENABLERANU:                    false
  ENU:                           true
  ERRORREPORTING:                false
  FEATURES:                      SQLENGINE, FULLTEXT
  FILESTREAMLEVEL:               0
  FILESTREAMSHARENAME:           <empty>
  FTSVCACCOUNT:                  NT Service\MSSQLFDLauncher$SPFARM
  FTSVCPASSWORD:                 <empty>
  HELP:                          false
  IACCEPTSQLSERVERLICENSETERMS:  true
  INDICATEPROGRESS:              false
  INSTALLSHAREDDIR:              C:\Program Files\Microsoft SQL Server\
  INSTALLSHAREDWOWDIR:           C:\Program Files (x86)\Microsoft SQL Server\
  INSTALLSQLDATADIR:             <empty>
  INSTANCEDIR:                   c:\SQL 2014\Instances
  INSTANCEID:                    SPFARM
  INSTANCENAME:                  SPFARM
  ISSVCACCOUNT:                  NT AUTHORITY\Network Service
  ISSVCPASSWORD:                 <empty>
  ISSVCSTARTUPTYPE:              Automatic
  MATRIXCMBRICKCOMMPORT:         0
  MATRIXCMSERVERNAME:            <empty>
  MATRIXNAME:                    <empty>
  NPENABLED:                     0
  PID:                           *****
  QUIET:                         true
  QUIETSIMPLE:                   false
  ROLE:                          
  RSINSTALLMODE:                 DefaultNativeMode
  RSSHPINSTALLMODE:              DefaultSharePointMode
  RSSVCACCOUNT:                  <empty>
  RSSVCPASSWORD:                 <empty>
  RSSVCSTARTUPTYPE:              Automatic
  SAPWD:                         <empty>
  SECURITYMODE:                  <empty>
  SQLBACKUPDIR:                  <empty>
  SQLCOLLATION:                  SQL_Latin1_General_CP1_CI_AS
  SQLSVCACCOUNT:                 NT Service\MSSQL$SPFARM
  SQLSVCPASSWORD:                <empty>
  SQLSVCSTARTUPTYPE:             Automatic
  SQLSYSADMINACCOUNTS:           <empty>
  SQLTEMPDBDIR:                  <empty>
  SQLTEMPDBLOGDIR:               <empty>
  SQLUSERDBDIR:                  <empty>
  SQLUSERDBLOGDIR:               <empty>
  SQMREPORTING:                  false
  TCPENABLED:                    1
  UIMODE:                        Normal
  UpdateEnabled:                 true
  UpdateSource:                  Slipstream
  X86:                           false

**Comment by johlju** [comment]
Okey, so I tested this on my dev environment now, and everything works. No bugs. It might be that you are missing something in your configuration file. Are you encrypting the credentials? Maybe the credentials are never stored in the resulting .mof file because you are not encrypting them? 

Below is the configuration I used, and the summary output. 
_Note: I stripped this down from a very large environment of 4 nodes, so I hope I got all in. :)_

Configuration + LCM configuration:
```
$ConfigData = @{
    AllNodes = @(
        @{
            NodeName= '*'

            # The path to the .cer file containing the public key of the certificate used to encrypt the credentials 
            CertificateFile = 'C:\Certs\MyCertForDSC.cer'
            
            # The thumbprint of the certificate used to encrypt the credentials. This tells LCM which certificate to use to decrypt the credentials. 
            # A certificate that is stored on the local machine store and identified in the LCM Configuration file
            Thumbprint = 'B6F48B3AE45AA2620332456B60DBAD1D3DB1A59E' 
            
            SqlTestInstanceName = 'TEST'
            
            # These values must be UPPERCASE.
            SqlInstallFeatures = 'SQLENGINE,SSMS' 

            # Location of Windows Server sxs folder.
            WindowsSourceSxs = '\\dev.company.local\DSC\Win2k12R2\Sources\Sxs'

            PSDscAllowDomainUser = $true
        }

        @{ 
            NodeName = 'sql.company.local'

            # Unique guid for this node (created random with New-Guid)
            Guid = '6c72b7fd-06c3-4b32-8405-99734879c84f'
        }
    )
}

[DSCLocalConfigurationManager()]
configuration LCMConfig 
{
    Node $AllNodes.NodeName
    {
        Settings
        {
            ConfigurationID = $node.Guid
            RefreshMode = 'Push'
            RebootNodeIfNeeded = $true
            ConfigurationMode = 'ApplyAndAutoCorrect'
            CertificateId = $node.Thumbprint 
            AllowModuleOverwrite = $true
        }
    }
}

Configuration SQLConfig
{
    param
    (
        [Parameter(Mandatory=$true)] 
        [ValidateNotNullorEmpty()] 
        [PsCredential]$SqlInstallCredential, 

        [Parameter(Mandatory=$false)] 
        [ValidateNotNullorEmpty()] 
        [PsCredential]$SqlAdministratorCredential = $SqlInstallCredential, 

        [Parameter(Mandatory=$true)] 
        [ValidateNotNullorEmpty()] 
        [PsCredential]$SqlServiceCredential, 

        [Parameter(Mandatory=$false)] 
        [ValidateNotNullorEmpty()] 
        [PsCredential]$SqlAgentServiceCredential = $SqlServiceCredential
    )

    Import-DscResource -ModuleName PSDesiredStateConfiguration
    Import-DscResource -ModuleName xSqlServer

    #region Primary Replica
    Node $AllNodes.NodeName
    {
        #region Install prerequisites for SQL Server
        WindowsFeature NET35 {
           Name = 'NET-Framework-Core'
           Source = $node.WindowsSourceSxs
           Ensure = 'Present'
        }

        WindowsFeature NET45 {
           Name = 'NET-Framework-45-Core'
           Ensure = 'Present'
        }
        #endregion Install prerequisites for SQL Server

        #region Install SQL Server (TEST-instance)
        xSQLServerSetup InstallSqlTestInstance
        {
            InstanceName = $Node.SqlTestInstanceName
            SetupCredential = $SqlInstallCredential
            Features= $Node.SqlInstallFeatures
            SourcePath = "\\dev\Binaries\SQL2014SP1"
            SQLSvcAccount = $SqlServiceCredential
            AgtSvcAccount = $SqlAgentServiceCredential
            ASSvcAccount = $SqlServiceCredential

            DependsOn = '[WindowsFeature]NET35','[WindowsFeature]NET45'
        }
        #endregion Install SQL Server (TEST-instance)
    }

}

# The following line invokes the LCM configuration and creates a .meta.mof file,  at the specified path, for each server in the $ConfigData hash table.
LCMConfig `
    -ConfigurationData $ConfigData `
    -OutputPath 'C:\DSC\Configuration'

# Get all the credentials needed
$SqlInstallCredential = Get-Credential -Message 'Enter credentials for SQL Setup account'
$SqlServiceCredential = Get-Credential -Message 'Enter credentials for SQL Service account'

# The following line invokes the DSC configuration and creates a .mof file,  at the specified path, for each server in the $ConfigData hash table.
SQLConfig `
    -SqlInstallCredential $SqlInstallCredential `
    -SqlServiceCredential $SqlServiceCredential `
    -ConfigurationData $ConfigData `
    -OutputPath 'C:\DSC\Configuration'
```

And this is the summary output.

```
User Input Settings:
  ACTION:                        Install
  ADDCURRENTUSERASSQLADMIN:      false
  AGTSVCACCOUNT:                 COMPANY\SqlSvc
  AGTSVCPASSWORD:                *****
  AGTSVCSTARTUPTYPE:             Automatic
  ASBACKUPDIR:                   Backup
  ASCOLLATION:                   Latin1_General_CI_AS
  ASCONFIGDIR:                   Config
  ASDATADIR:                     Data
  ASLOGDIR:                      Log
  ASPROVIDERMSOLAP:              1
  ASSERVERMODE:                  MULTIDIMENSIONAL
  ASSVCACCOUNT:                  <empty>
  ASSVCPASSWORD:                 <empty>
  ASSVCSTARTUPTYPE:              Automatic
  ASSYSADMINACCOUNTS:            <empty>
  ASTEMPDIR:                     Temp
  BROWSERSVCSTARTUPTYPE:         Automatic
  CLTCTLRNAME:                   <empty>
  CLTRESULTDIR:                  <empty>
  CLTSTARTUPTYPE:                0
  CLTSVCACCOUNT:                 <empty>
  CLTSVCPASSWORD:                <empty>
  CLTWORKINGDIR:                 <empty>
  COMMFABRICENCRYPTION:          0
  COMMFABRICNETWORKLEVEL:        0
  COMMFABRICPORT:                0
  CONFIGURATIONFILE:             C:\Program Files\Microsoft SQL Server\120\Setup Bootstrap\Log\20160824_174619\ConfigurationFile.ini
  CTLRSTARTUPTYPE:               0
  CTLRSVCACCOUNT:                <empty>
  CTLRSVCPASSWORD:               <empty>
  CTLRUSERS:                     <empty>
  ENABLERANU:                    false
  ENU:                           true
  ERRORREPORTING:                false
  FEATURES:                      SQLENGINE
  FILESTREAMLEVEL:               0
  FILESTREAMSHARENAME:           <empty>
  FTSVCACCOUNT:                  <empty>
  FTSVCPASSWORD:                 <empty>
  HELP:                          false
  IACCEPTSQLSERVERLICENSETERMS:  true
  INDICATEPROGRESS:              false
  INSTALLSHAREDDIR:              C:\Program Files\Microsoft SQL Server\
  INSTALLSHAREDWOWDIR:           C:\Program Files (x86)\Microsoft SQL Server\
  INSTALLSQLDATADIR:             <empty>
  INSTANCEDIR:                   C:\Program Files\Microsoft SQL Server\
  INSTANCEID:                    TEST
  INSTANCENAME:                  TEST
  ISSVCACCOUNT:                  NT AUTHORITY\Network Service
  ISSVCPASSWORD:                 <empty>
  ISSVCSTARTUPTYPE:              Automatic
  MATRIXCMBRICKCOMMPORT:         0
  MATRIXCMSERVERNAME:            <empty>
  MATRIXNAME:                    <empty>
  NPENABLED:                     0
  PID:                           *****
  QUIET:                         true
  QUIETSIMPLE:                   false
  ROLE:                          
  RSINSTALLMODE:                 DefaultNativeMode
  RSSHPINSTALLMODE:              DefaultSharePointMode
  RSSVCACCOUNT:                  <empty>
  RSSVCPASSWORD:                 <empty>
  RSSVCSTARTUPTYPE:              Automatic
  SAPWD:                         <empty>
  SECURITYMODE:                  <empty>
  SQLBACKUPDIR:                  <empty>
  SQLCOLLATION:                  Finnish_Swedish_CI_AS
  SQLSVCACCOUNT:                 COMPANY\SqlSvc
  SQLSVCPASSWORD:                *****
  SQLSVCSTARTUPTYPE:             Automatic
  SQLSYSADMINACCOUNTS:           COMPANY\SqlUser
  SQLTEMPDBDIR:                  <empty>
  SQLTEMPDBLOGDIR:               <empty>
  SQLUSERDBDIR:                  <empty>
  SQLUSERDBLOGDIR:               <empty>
  SQMREPORTING:                  false
  TCPENABLED:                    1
  UIMODE:                        Normal
  UpdateEnabled:                 true
  UpdateSource:                  Slipstream
  USEMICROSOFTUPDATE:            false
  X86:                           false

```

**Comment by AlexandrosIV** [comment]
I am using this simple script
`
configuration SimpleSpFarm
{
     
    Import-DscResource -ModuleName xSQLServer -ModuleVersion 1.8.0.0
 
    node b7e27e5e-bc59-42f9-b75c-61e122bf8533 {

        $cred = Get-Credential sparta\sp_admin
      
        xSQLServerSetup StandAloneInstall {

            InstanceName = 'SPFarm'
            SetupCredential = $cred
            Features =   'SQLEngine,FullText'
            SourcePath  = '\\MYWINDOWSVM01\Binaries\SQL2012SP2\'
            SQLSvcAccount = $cred
            InstanceDir = 'c:\SQL 2014\Instances'
            InstallSharedDir = 'c:\SQL 2014\Instances'
            InstallSQLDataDir = 'c:\SQL 2014\Binaries'
            SQLUserDBDir = 'c:\SQL 2014\DataFiles'
            SQLUserDBLogDir = 'c:\SQL 2014\Logs'
            SQLTempDBDir = 'c:\SQL 2014\Temp'
            SQLTempDBLogDir = 'c:\SQL 2014\Temp'
            SQLSysAdminAccounts = 'sparta\sp_admin'
            AgtSvcAccount = $cred
            ASSvcAccount = $cred  
        }

    }
}

$configData = @{
    AllNodes = @(
        @{
            NodeName = "b7e27e5e-bc59-42f9-b75c-61e122bf8533"
            PSDscAllowDomainUser = $True
            PsDscAllowPlainTextPassword = $True
            }
       
    )
}

SimpleSpFarm -OutputPath 'C:\DSC scripts\SimpleSpFarm' -ConfigurationData $configData

in the .mof generated, I see the credentials stored

`/*
@TargetNode='b7e27e5e-bc59-42f9-b75c-61e122bf8533'
@GeneratedBy=sp_admin
@GenerationDate=08/24/2016 18:40:38
@GenerationHost=MYWINDOWSVM01
*/

instance of MSFT_Credential as $MSFT_Credential1ref
{
Password = "AAaa11112222";
 UserName = "sparta\\sp_admin";

};

instance of MSFT_Credential as $MSFT_Credential2ref
{
Password = "AAaa11112222";
 UserName = "sparta\\sp_admin";

};

instance of MSFT_Credential as $MSFT_Credential3ref
{
Password = "AAaa11112222";
 UserName = "sparta\\sp_admin";

};

instance of MSFT_Credential as $MSFT_Credential4ref
{
Password = "AAaa11112222";
 UserName = "sparta\\sp_admin";

};

instance of MSFT_xSQLServerSetup as $MSFT_xSQLServerSetup1ref
{
SQLUserDBLogDir = "c:\\SQL 2014\\Logs";
 SQLTempDBDir = "c:\\SQL 2014\\Temp";
 AgtSvcAccount = $MSFT_Credential1ref;
 SQLUserDBDir = "c:\\SQL 2014\\DataFiles";
 ResourceID = "[xSQLServerSetup]StandAloneInstall";
 ASSvcAccount = $MSFT_Credential2ref;
 SQLSysAdminAccounts = {
    "sparta\\sp_admin"
};
 InstallSQLDataDir = "c:\\SQL 2014\\Binaries";
 SQLTempDBLogDir = "c:\\SQL 2014\\Temp";
 SetupCredential = $MSFT_Credential3ref;
 InstanceName = "SPFarm";
 SourceInfo = "C:\\DSC scripts\\SimpleSpFarm\\SimpleFarm.ps1::25::9::xSQLServerSetup";
 InstanceDir = "c:\\SQL 2014\\Instances";
 SQLSvcAccount = $MSFT_Credential4ref;
 SourcePath = "\\\\MYWINDOWSVM01\\Binaries\\SQL2012SP2\\";
 ModuleVersion = "1.8.0.0";
 ModuleName = "xSQLServer";
 Features = "SQLEngine,FullText";
 InstallSharedDir = "c:\\SQL 2014\\Instances";

 ConfigurationName = "SimpleSpFarm";

};
instance of OMI_ConfigurationDocument


                    {
 Version="2.0.0";
 

                        MinimumCompatibleVersion = "1.0.0";
 

                        CompatibleVersionAdditionalProperties= {"Omi_BaseResource:ConfigurationName"};
 

                        Author="sp_admin";
 

                        GenerationDate="08/24/2016 18:40:38";
 

                        GenerationHost="MYWINDOWSVM01";
 

                        Name="SimpleSpFarm";


                    };
`
`

could be an issue with storing in plaintext??
the weird thing is that the first test I did with the resource it succeeded, but not sure what I did different, cant recall. 
I am just starting with DSC so is probably that the error is in my side, but I cant see where

**Comment by AlexandrosIV** [comment]
Definetively not a Bug

your code worked fine, but mine still failing
Your code
` xSQLServerSetup InstallSqlTestInstance
        {
            InstanceName = $Node.SqlTestInstanceName
            SetupCredential = $SqlInstallCredential
            Features= $Node.SqlInstallFeatures
            SourcePath = "\\MyWindowsVM01\Binaries\SQL2014SP2\"
            SQLSvcAccount = $SqlServiceCredential
            AgtSvcAccount = $SqlAgentServiceCredential
            ASSvcAccount = $SqlServiceCredential

        }
        #endregion Install SQL Server (TEST-instance)
    }

}
# Get all the credentials needed
$SqlInstallCredential = Get-Credential -Message 'Enter credentials for SQL Setup account'
$SqlServiceCredential = Get-Credential -Message 'Enter credentials for SQL Service account'`

my code

` node SQL_backend {

        xSQLServerSetup SQLInstance {

            Features = 'SQLEngine,SSMS'
            SetupCredential = $cred
            InstanceName = 'SQL2014'
            SourcePath = '\\MyWindowsVM01\Binaries\SQL2014SP2\'

        }
      
    }

}
$cred = Get-Credential sparta\sp_admin`

cant see any difference.
I tried with the credentials in plain text, or using a certificate, still the same.
I see the the credentials stored  in the .mof, 
`/*
@TargetNode='SQL_backend'
@GeneratedBy=sp_admin
@GenerationDate=08/25/2016 19:53:13
@GenerationHost=MYWINDOWSVM01
*/

instance of MSFT_Credential as $MSFT_Credential1ref
{
Password = "-----BEGIN CMS-----\nMIIBnAYJKoZIhvcNAQcDoIIBjTCCAYkCAQAxggFEMIIBQAIBADAoMBQxEjAQBgNVBAMMCVdJTkRP\nV1MwNAIQGSNd1xMEGYFIrHR84kL8LjANBgkqhkiG9w0BAQcwAASCAQDTU2QKLpF810AD2UUYeefR\nEWR3vXfoc3gS/RVauvXtpFWijVEyIw2zAVZnihpf0rMiUM5k8ZJGHGVnQqWz1uYmfrF3IUhhpiu5\nPLGExBSMJgqKJX0C5IkpQXtoZMeJ+G4vq420BfdVdMewXEMMn9d1zIrypk4ECOrOrC7wie59e09e\nWRg9SwQnvngVZmKRxruJOQiZNtH5dqZRkVvko1goAocani8RBjX0fvr6gr+qDgS9ZPYJ3Oe1nHq2\nTVudTt7z3Grtisu0IEhOzmt1g0aZy4TtjsrVwPZEfuNTlkxNbmir0hYZOiWErPm5E6WfWu4jvXVi\nEoI/opBI27zxJcxFMDwGCSqGSIb3DQEHATAdBglghkgBZQMEASoEEFXpr5hMrNHBjegdzShOy7mA\nEKf6DYi7/orwHRWJsNYHA2s=\n-----END CMS-----";
 UserName = "sparta\\sp_admin";

};

instance of MSFT_xSQLServerSetup as $MSFT_xSQLServerSetup1ref
{
SourcePath = "\\\\MyWindowsVM01\\Binaries\\SQL2014SP2\\";
 InstanceName = "SQL2014";
 SetupCredential = $MSFT_Credential1ref;
 SourceInfo = "::9::9::xSQLServerSetup";
 ResourceID = "[xSQLServerSetup]SQLInstance";
 ModuleName = "xSQLServer";
 ModuleVersion = "1.8.0.0";
 Features = "SQLEngine,SSMS";

 ConfigurationName = "SQLTest";

};
instance of OMI_ConfigurationDocument


                    {
 Version="2.0.0";
 

                        MinimumCompatibleVersion = "1.0.0";
 

                        CompatibleVersionAdditionalProperties= {"Omi_BaseResource:ConfigurationName"};
 

                        Author="sp_admin";
 

                        GenerationDate="08/25/2016 19:53:13";
 

                        GenerationHost="MYWINDOWSVM01";


                        ContentType="PasswordEncrypted";
 

                        Name="SQLTest";


                    };
`

but when executing the configuration in the node I see this in the call to the installer
my code
`VERBOSE:[WINDOWS04]:                            [[xSQLServerSetup]SQLInstance] Arguments: /Quiet="True" /IAcceptSQLServerLicenseTerms="True" /Action="Install
" /InstanceName="SQL2014" /Features="SQLEngine,SSMS"` 

your code

`VERBOSE: [WINDOWS04]:                            [[xSQLServerSetup]InstallSqlTestInstance] Arguments: /Quiet="True" /IAcceptSQLServerLicenseTerms="True" /Acti
on="Install" /SQLSVCACCOUNT="sparta\sql_service" /SQLSVCPASSWORD="********" /AGTSVCACCOUNT="sparta\sql_service" /AGTSVCPASSWORD="********" /AGTSVCSTARTUPTYPE=
Automatic /InstanceName="TEST" /Features="SQLENGINE,SSMS" /SQLSysAdminAccounts="sparta\sp_admin"
`

really this is driving me nuts, it must be a simple thing, but cant see it

**Comment by johlju** [comment]
Weird. Would also like to know why this happens. Could you upload your actual code to a personal (yours) public repository? Then I can test your actual code against my environment. I'm gonna be away this weekend so I won't be able to test the code until next week. But I probably will glance at it before then to see if I see anything.

_Just create a repository in your account and just drag the files in. Replace any sensitive computer names, accounts etc. to dummy ones before uploading/dragging. :)_

**Comment by johlju** [comment]
Can you try make the features upper-case? 
Change: `Features = 'SQLEngine,SSMS'`
To: `Features = 'SQLENGINE,SSMS'`

It's a bug that these are not forced to upper case in the code. See PR #46.

**Comment by AlexandrosIV** [comment]
Its working now...

The solution is Features = 'SQLENGINE,SSMS' in Upper Case as you suggested...

:S

**Comment by johlju** [comment]
I hope we can get those to be "case-insensitive" soon. 

Let's close this issue of and track this bug in issue #38 instead. And make sure issue #38 gets fixed in PR #46 (currently it does not).
